authors/apps/articles/views.py

```diff
@@ -20,15 +24,31 @@ class ListCreateArticleAPIView(APIView):
 
     def post(self, request):
         article = request.data.get('article', {})
+
         user_data = {"author": request.user.profile}
         serializer = self.serializer_class(data=article, context=user_data)
+        serializer_context = {
+            "author": request.user.profile,
+            'request': request
+        }
+        serializer = self.serializer_class(data=article, context=serializer_context)
```

Comment:
Hi @reifred, this is good work. I just need some clarification on something. I have noticed that you are initializing the serializer twice in this bit of code. Why not delete 
 ```
user_data = {"author": request.user.profile}
serializer = self.serializer_class(data=article, context=user_data)
```
and keep 
```
serializer_context = {
            "author": request.user.profile,
            'request': request
        }
serializer = self.serializer_class(data=article, context=serializer_context)
```
since they do the same thing, the only difference being the addition of the `request` to the serializer context. Also note that you did not really need to create the `serializer_context` dictionary since the `request` could have been added to the `user_data` dictionary with the same outcome achieved.

Reply:
Thanks for this @richien let me work on it.